microsoft/testsuites/core/storage.py

```diff
@@ -374,6 +376,63 @@ def verify_azure_file_share_nfs(self, log: Logger, node: Node) -> None:
         finally:
             nfs.delete_share()
             node.tools[NFSClient].stop(mount_dir)
+    
+    _file_path = "~/SAMPLE.txt"
+    _original_text = "Hello, world!"
+    _mount_point = "/mnt/data0"
+
+    @TestCaseMetadata(
+        description="""
+        This test case will verify mount azure nfs on guest successfully.
+        1. Write “Hello World!” string to a file. 
+        2. Move the file from home folder to data disk. 
+        3. Assert the file content in data disk. 
+        4. Assert the file on original place doesn't exist. 
+        """,
+        timeout=TIME_OUT,
+        priority=4,
+        requirement=simple_requirement(
+            disk=AzureDiskOptionSettings(
+                data_disk_count=search_space.IntRange(min=1),
+                data_disk_size=search_space.IntRange(min=20),
+            ),supported_platform_type=[AZURE],
+        ),
+    )
+    def check_disk_write_operation_ripastri(self, log: Logger, node: RemoteNode) -> None:
+        # write content to the file
+        node.tools[Echo].write_to_file(
+            self._original_text, node.get_pure_path(self._file_path), sudo=True
+        )
+
+        # read content from the file
+        read_text = node.tools[Cat].read(self._file_path, force_run=True, sudo=True)
+        
+        assert_that(
+            read_text,
+            f"content read from file {self._file_path}: '{read_text}' should be equal to content written to file '{self._original_text}'",
+        ).is_equal_to(self._original_text)
+
+        # mount the data disk
+        data_disks = node.features[Disk].get_raw_data_disks()
+
+        assert_that(data_disks, f"Unable to find required data_disks.").is_not_empty()
+
+        node.tools[Mount].mount(data_disks[0], self._mount_point, format=True)
+        dest_path = node.get_pure_path(self._mount_point).joinpath("sample.txt") 
+        
+        # Move file to data disk
+        node.tools[Mv].move_file(self._file_path, dest_path.as_posix(), sudo=True)
+
+        # read content from the file after moving
+        read_text = node.tools[Cat].read(dest_path.as_posix(), force_run=True, sudo=True)
+        
+        assert_that(
+            read_text,
+            f"content read from the file {dest_path}: '{read_text}' should be equal to content written to file: '{self._original_text}'",
```

Comment:
[nit] We don't need to mention the compared quantities in assert_that message, and it should be displayed by assertion error message. We can remove `read_text` and `_original_text`

Reply:
Removed as requested.